src/main/java/Duke.java

```diff
@@ -1,10 +1,82 @@
+                        naruto.printBetweenBars("[" + taskList.getStatusIcon(taskNumber) + "] "
+                                + taskList.getDescription(taskNumber));
+                        storage.save(taskList.getUpdatedTasks());
+                        break;
+                    case TODO:
+                        try {
+                            taskNumber = taskList.addToDo(input);
+                            naruto.printTaskAddedMessage("[" + taskList.getTaskIcon(taskNumber) + "]["
+                                    + taskList.getStatusIcon(taskNumber) + "] "
+                                    + taskList.getDescription(taskNumber), taskNumber);
+                            storage.save(taskList.getUpdatedTasks());
+                        } catch (DukeException dE) {
+                            naruto.say("☹ OOPS!!! The description of a todo cannot be empty.");
+                        }
+                        break;
+                    case DEADLINE:
+                        taskNumber = taskList.addDeadline(input);
+                        naruto.printTaskAddedMessage("[" + taskList.getTaskIcon(taskNumber) + "]["
+                                + taskList.getStatusIcon(taskNumber) + "] "
+                                + taskList.getDescription(taskNumber), taskNumber);
+                        storage.save(taskList.getUpdatedTasks());
+                        break;
+                    case EVENT:
+                        taskNumber = taskList.addEvent(input);
+                        naruto.printTaskAddedMessage("[" + taskList.getTaskIcon(taskNumber) + "]["
+                                + taskList.getStatusIcon(taskNumber) + "] "
+                                + taskList.getDescription(taskNumber), taskNumber);
+
+                        storage.save(taskList.getUpdatedTasks());
+                        break;
+                    case DELETE:
+                        taskNumber = Integer.parseInt(input.substring("delete".length() + 1)) - 1;
+                        naruto.say("Noted. I've removed this task");
+                        naruto.printBetweenBars("[" + taskList.getTaskIcon(taskNumber) + "]["
+                                + taskList.getStatusIcon(taskNumber) + "] " + taskList.getDescription(taskNumber));
+                        naruto.say("Now you have " + (taskList.getTaskCount() - 1) + " tasks in the list");
+                        taskList.delete(taskNumber);
+                        storage.save(taskList.getUpdatedTasks());
+                        break;
+                    case FIND:
+                        // Test merging
+                        naruto.printMatchingItems(taskList.getUpdatedTasks(), input.substring("find".length() + 1));
+                        break;
+                }
+            } catch (DukeException dE) {
+                naruto.say("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+        }
```

Comment:
The main method is abit long. Can define the necessary classes as variables for the duke class, 

i.e. private Ui naruto = new Ui("Naruto"); as opposed to defining them in the main method. 

Also, can try to break up the different cases into methods, such as 

case DELETE: 
delete(); 
break; 

and all your relevant code inside the delete function for easier debugging next time!